### PR TITLE
Nextjs: Improve Google Fonts failure error messages and documentation

### PR DESCRIPTION
**Pull request by nsheaps**
Helpful for debugging which fonts need to be mocked

related: https://github.com/storybookjs/storybook/issues/23332

## What I did

Made the error message (which I saw typically thrown during chromatic builds) useful by including the original error in the newly thrown error, which makes the mocking tasks way easier.

Documentation also added for mocking (which is really part of nextjs but no documentation exists)

## How to test

Tested by manually adding these changes in minified code when testing mocked font responses for use with chromatic.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/writing-stories/build-pages-with-storybook.md
Comment (nsheaps) [inline docs/writing-stories/build-pages-with-storybook.md:398]: Not sure if this is the right place to put this, but it seems like this file is all about mocks

Comment (yannbf) [reply docs/writing-stories/build-pages-with-storybook.md:398]: @kylegach are you saying @nsheaps should move all of the changes into that README.md file? Or maybe in the [FAQ](https://storybook.js.org/docs/react/faq)?

Comment (kylegach) [reply docs/writing-stories/build-pages-with-storybook.md:398]: Yes, into the README. For now. We'll base the better-located docs on those READMEs when the time comes.

Comment (nsheaps) [reply docs/writing-stories/build-pages-with-storybook.md:398]: I'll take care of that in the upcoming days, thanks for the feedback!

File: code/frameworks/nextjs/README.md
Comment (yannbf) [inline code/frameworks/nextjs/README.md:278]:
```suggestion
Occasionally fetching fonts from Google may fail as part of your Storybook build step. It is highly recommended to mock these requests, as those failures can cause your pipeline to fail as well. Next.js [supports mocking fonts](https://github.com/vercel/next.js/blob/725ddc7371f80cca273779d37f961c3e20356f95/packages/font/src/google/fetch-css-from-google-fonts.ts#L36) via a JavaScript module located where the env var `NEXT_FONT_GOOGLE_MOCKED_RESPONSES` references.

```

File: code/frameworks/nextjs/README.md
Comment (yannbf) [inline code/frameworks/nextjs/README.md:291]:
```suggestion
Your mocked fonts will look something like this:
```

**Review by jonniebigodes — APPROVED** [review]
@nsheaps thanks for addressing the feedback so promptly. Appreciate it 🙏 ! Documentation-wise it's all good on my end. I'll defer to the other maintainers and see what they have to say about it.

Hope you have a great day.

Stay safe

**Comment by yannbf** [comment]
> @nsheaps thanks for addressing the feedback so promptly. Appreciate it 🙏 ! Documentation-wise it's all good on my end. I'll defer to the other maintainers and see what they have to say about it.
> 
> Hope you have a great day.
> 
> Stay safe

I see some opportunity to use the categorized errors framework here, so leave it to me and I'll update this branch soon and merge it!

File: code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts
Comment (nsheaps) [inline code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts:5]: I think there might be a typo here, Goggle vs Google

Comment (yannbf) [reply code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts:5]: Well spotted! Thank you <3

**Pull request merged**
